File: run_test_print.py
import time

if __name__ == "__main__":
    print_range = 10000000

    start_time = time.time()

    for i in range(print_range + 1):
        print(f"\r{i}/{print_range}", end="")

    end_time = time.time()
    elapsed_time = end_time - start_time
    print(f"\nThe program execution time: {elapsed_time} seconds")

File: run_test_print.lua
--[luarocks install luasocket]--
local time = require("socket")

function main()
    local print_range = 10000000

    local start_time = time.gettime()

    for i = 1, print_range do
        io.write(("\r%d/%d"):format(i, print_range))
        io.flush()
    end

    local end_time = time.gettime()
    local elapsed_time = end_time - start_time
    print(("\nThe program execution time: %.2f seconds"):format(elapsed_time))
end

main()